File: Ruby/2_blink.rb
#!/usr/bin/env ruby

require 'serialport'

def blink(port)
  serial = SerialPort.new("#{port}", 9600, 8, 1)
  serial.read_timeout = 5000
  led_on = false

  loop do
    led_on = !led_on
    if led_on
      serial.write "D1"
    else
      serial.write "D0"
    end

    if led_on
      state = "On"
    else
      state = "Off"
    end
    print "LED is #{state}\r"
    STDOUT.flush
    sleep(1)
  end
end


if __FILE__ == $0
  if ARGV.empty?
    puts "usage: ruby #{__FILE__} <serial>"
  else
    blink(ARGV[0])
  end
end

File: Python/2_blink.py
# -*- coding: utf-8 -*-
import time
import sys
from serial import Serial

def main():
    if len(sys.argv) != 2:
        print "usage: python %s <serial>" % __file__
        sys.exit(2)

    serial = Serial(sys.argv[1])
    led_on = False

    while True:
        led_on = not led_on
        if led_on:
            serial.write("D1")
        else:
            serial.write("D0")
        state = 'On' if led_on else 'Off'
        sys.stdout.write("LED is %s \r" % state)
        sys.stdout.flush()
        time.sleep(1)

if __name__ == '__main__':
    main()
